File: classes/written_response.py
import datetime

from enums import DebuggingStage

class WrittenResponse:

    def __init__(self, exercise_name: str = None, exercise_description: str = None, debugging_stage: DebuggingStage = None, end_time: datetime = None, response: str = None):
        self._exercise_name: str = exercise_name
        self._exercise_description: str = exercise_description
        self._debugging_stage: DebuggingStage = debugging_stage
        self._end_time: datetime = end_time
        self._response: str = response

    @property
    def exercise_name(self) -> str:
        return self._exercise_name

    @property
    def exercise_description(self) -> str:
        return self._exercise_description

    @property
    def debugging_stage(self) -> DebuggingStage:
        return self._debugging_stage

    @property
    def end_time(self) -> datetime:
        return self._end_time

    @property
    def response(self) -> str:
        return self._response

File: written_response_analysis/written-response.ts
import { DebuggingStage } from "./debugging-stage.ts";

export class WrittenResponse {

    private exercise: string;
    private exerciseDescription: string;
    private debuggingStage: DebuggingStage;
    private response: string;
    private code: string | null;

    constructor(exercise: string, exerciseDescription: string, debuggingStage: string, response: string, code: string | null = null) {
        this.response = response;
        this.exerciseDescription = exerciseDescription;
        this.debuggingStage = DebuggingStage[debuggingStage as keyof typeof DebuggingStage];
        this.exercise = exercise;
        this.code = code;
    }

    getExercise(): string {
        return this.exercise;
    }

    getExerciseDescription(): string {
        return this.exerciseDescription;
    }

    getDebuggingStage(): DebuggingStage {
        return this.debuggingStage;
    }

    getResponse(): string {
        return this.response;
    }
}